File: src/main/webapp/js/service/CashLogService.js
//服务层
app.service('cashLogService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../cashLog/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../cashLog/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../cashLog/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../cashLog/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../cashLog/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../cashLog/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../cashLog/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
});

File: src/main/java/com/wyan/service/CashLogService.java
package com.wyan.service;

import com.github.pagehelper.PageInfo;
import com.wyan.pojo.TbCashLog;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CashLogService {

	/**
	 * 返回全部列表
	 * @return
     * @param cashLog
	 */
	public List<TbCashLog> findAll(TbCashLog cashLog);
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbCashLog cash_log) throws Exception;


	/**
	 * 修改
	 */
	public void update(TbCashLog cash_log);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbCashLog findOne(Long id);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbCashLog cash_log, int pageNum, int pageSize);

	Long findCostMoney();

	Long findMonthCostMoney();

	Long findTotalCostMoney();

	Long yearCostMoney();

	void updateState(TbCashLog cashLog);

	void updateImg(TbCashLog cashLog);
}
